serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -114,8 +112,13 @@ def add(self, request_states: list[RequestState]):
                 if (
                     request_state.validation_err is not None
                     or request_state.prompt_len >= self.max_context_length
+                    or request_state.prompt_len >= self.max_num_batched_tokens > 0
```

Comment:
Let's clean up the two conditions as 
`request_state.prompt_len >= min(self.max_context_length, self.max_num_batched_tokens)`

Reply:
I am confused that we have a comparison of `self.max_num_batched_tokens > 0`. It means that at the moment when we added such condition, we were not sure if max_num_batched_tokens can be zero. Are we sure that `max_num_batched_tokens` is always positive?

Reply:
`max_num_batched_tokens` is of course always positive. I thought `> 0` was a typo.

Reply:
fixed

Reply:
> max_num_batched_tokens is of course always positive. I thought > 0 was a typo.

It appeared not true. At least the unit test assumed that `max_num_batched_tokens` can be uninitialized and was equal to -1

Reply:
and due this reason we [have comparison](https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/engine/sync_engine.py#L276) on positive value in current `_adjust_batch`

@yelite @sunggg We need to decide if we can have `max_num_batched_tokens` always defined or not

Reply:
I can fix test and add assert into StagingInferenceEngine/SynchronousInferenceEngine initialization. Just need to understand if we see any inference Model that is not limited by maximum number of tokens

Reply:
ok, it seems it was something initially for testing, but it cannot work since logic of engines depends on certain assumptions. Eventually I removed ability not to check configuration in [base.py:42](https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/engine/base.py#L42) and modified test to pass verification. All tests passed

Reply:
had to rollback changes back because for context we have not less, but for `max_num_batched_tokens` we have to exclude only in case of exceeding. Boarding conditions are processed wrong if we have only `min(...)`